src/main/java/com/metabug/web/controller/IndexController.java

```diff
@@ -16,9 +34,35 @@ public ModelAndView index() {
     }
 
     @PostMapping(value = {"/", "/login"})
-    public ModelAndView login() {
+    public ModelAndView login(@Valid final LoginDto loginDto,
+                              final BindingResult bindingResult,
+                              final HttpServletRequest request) {
         final ModelAndView modelAndView = new ModelAndView();
 
+        if(bindingResult.hasErrors()) {
+            addWarning(modelAndView);
+            modelAndView.setViewName("index");
+            return modelAndView;
+        }
+
+        final User user = userService.findUserByLogin(loginDto.getLogin());
+
+        if(user != null) {
+            if(bCryptPasswordEncoder.matches(loginDto.getPassword(), user.getPassword())) {
```

Comment:
Maybe remove field `bCryptPasswordEncoder` from class and add method in `UserService` which will compare passwords using it's `bCryptPasswordEncoder` field?

Reply:
done